File: Java/Delete-Node-in-the-Middle-of-Singly-Linked-List.java
/**
 * Definition for ListNode.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int val) {
 *         this.val = val;
 *         this.next = null;
 *     }
 * }
 */ 
public class Solution {
    /**
     * @param node: the node in the list should be deleted
     * @return: nothing
     */
    public void deleteNode(ListNode node) {
        // write your code here
        if (node == null) return;
        while (node.next != null) {
        	node.val = node.next.val;
        	if (node.next.next == null) node.next = null;
        	else node = node.next;
        }
    }
}

File: Cpp/Delete-Node-in-the-Middle-of-Singly-Linked-List.cpp
/**
 * Definition of ListNode
 */
class ListNode {
public:
    int val;
    ListNode *next;
    ListNode(int val) {
        this->val = val;
        this->next = NULL;
    }
};

class Solution {
public:
    /**
     * @param node: a node in the list should be deleted
     * @return: nothing
     */
    void deleteNode(ListNode *node) {
        // write your code here
        if (node == NULL) return;
        while (node->next != NULL) {
            node->val = node->next->val;
            if (node->next->next == NULL) node->next = NULL;
            else node = node->next;
        }
    }
};
